source/scripts/InjectedScript/stargazerChainProviderProxy.ts

```diff
@@ -0,0 +1,85 @@
+import {
+  StargazerEncodedProxyRequest,
+  StargazerProxyRequest,
+  StargazerEncodedProxyResponse,
+  StargazerProxyResponse,
+} from '../common/proxy-types';
+
+import { StargazerChainProviderError, StargazerChainProviderRpcError } from './errors';
+import { genProxyReqId } from './utils';
+
+class StargazerChainProviderProxy {
+  retreiveProxyOnce() {
+    const scriptElem = document.querySelector('[data-stargazer-injected=injected]');
+    if (!scriptElem || !(scriptElem instanceof HTMLScriptElement)) {
+      throw new StargazerChainProviderError('Unable to retreive proxy once element');
+    }
+
+    if (!scriptElem.dataset.stargazerOnce) {
+      throw new StargazerChainProviderError('Unable to retreive proxy once attribute');
+    }
+
+    return scriptElem.dataset.stargazerOnce;
+  }
+
+  request<T = any>(request: StargazerProxyRequest): Promise<T> {
+    return new Promise<T>((rs, rj) => {
```

Comment:
```suggestion
    return new Promise<T>((resolve, reject) => {
```
No non-standard two letter variables please.